index.js

```diff
@@ -242,16 +234,16 @@ module.exports = function (filename, opts) {
       debug("data doesn't fit current block, creating new")
     }
 
-    appendRecord(latestBlockBuf, encodedData, nextOffsetInBlock)
+    Record.write(latestBlockBuf, nextOffsetInBlock, encodedData)
     cache.set(latestBlockIndex, latestBlockBuf) // update cache
-    const offset = nextOffsetInBlock + latestBlockIndex * blockSize
-    nextOffsetInBlock += recordSize(encodedData)
+    const offset = latestBlockIndex * blockSize + nextOffsetInBlock
     blocksToBeWritten.set(latestBlockIndex, {
       blockBuf: latestBlockBuf,
       offset,
     })
     scheduleWrite()
     debug('data inserted at offset %d', offset)
+    nextOffsetInBlock += Record.size(encodedData)
```

Comment:
can we leave this were it was? That way we set blocksToBeWritten & nextOffsetInBlock and only after that scheduleWrite